src/drt/src/io/io.cpp

```diff
@@ -417,8 +417,11 @@ void io::Parser::createNDR(odb::dbTechNonDefaultRule* ndr)
   std::vector<odb::dbTechVia*> vias;
   ndr->getUseVias(vias);
   for (auto via : vias) {
-    fnd->addVia(design_->getTech()->getVia(via->getName()),
-                via->getBottomLayer()->getNumber() / 2);
+    z = design_->tech_->getLayer(via->getBottomLayer()->getName())
+                ->getLayerNum()
+            / 2
+        - 1;
```

Comment:
The formatting is terrible - please split out `via->getBottomLayer()->getName()` so this can be condensed.

Where does the -1 come from?

Reply:
The -1 is related to the master slice layer. The z-coordinates start from the first routing layer with zIdx = 0, while getLayerNum returns the layer number for the first routing layer as 2.